### ISSUE TITLE
nvme/033 failed

### ISSUE THREAD

**Issue by zhijianli88** [osandov/blktests#92]
```
[root@rdma-client blktests]# TEST_DEVS=/dev/nvme0n1 nvme_trtype=rdma ./check nvme/033
nvme/033 => nvme0n1 (create and connect to an NVMeOF target with a passthru controller) [failed]
    runtime  2.562s  ...  2.561s
    --- tests/nvme/033.out      2022-05-11 12:06:52.956599746 +0800
    +++ /home/lizhijian/blktests/results/nvme0n1/nvme/033.out.bad       2022-06-02 11:55:51.772802585 +0800
    @@ -1,7 +1,10 @@                                                                                               
     Running nvme/033                                    
    -Discovery Log Number of Records 1, Generation counter X
    +Discovery Log Number of Records 2, Generation counter X
     =====Discovery Log Entry 0======
    -trtype:  loop                                       
    +trtype:  rdma                                       
    +subnqn:  nqn.2014-08.org.nvmexpress.discovery
    ...                                                  
    (Run 'diff -u tests/nvme/033.out /home/lizhijian/blktests/results/nvme0n1/nvme/033.out.bad' to see the entire diff)
```

the output of discovery as below:
```
[root@rdma-client blktests]# nvme discover -t rdma -a 192.168.122.78 -s 4420

Discovery Log Number of Records 2, Generation counter 2
=====Discovery Log Entry 0======
trtype:  rdma
adrfam:  ipv4
subtype: unrecognized
treq:    not specified, sq flow control disable supported
portid:  0
trsvcid: 4420
subnqn:  nqn.2014-08.org.nvmexpress.discovery
traddr:  192.168.122.78
rdma_prtype: not specified
rdma_qptype: connected
rdma_cms:    rdma-cm
rdma_pkey: 0x0000
=====Discovery Log Entry 1======
trtype:  rdma
adrfam:  ipv4
subtype: nvme subsystem
treq:    not specified, sq flow control disable supported
portid:  0
trsvcid: 4420
subnqn:  blktests-subsystem-1
traddr:  192.168.122.78
rdma_prtype: not specified
rdma_qptype: connected
rdma_cms:    rdma-cm
rdma_pkey: 0x0000
```

**Comment by zhijianli88** [comment]
Although change trtype to loop, it still failed
```
[root@rdma-client blktests]# TEST_DEVS=/dev/nvme0n1 ./check nvme/033 
nvme/033 => nvme0n1 (create and connect to an NVMeOF target with a passthru controller) [failed]
    runtime  2.173s  ...  2.145s
    --- tests/nvme/033.out      2022-05-11 12:06:52.956599746 +0800
    +++ /home/lizhijian/blktests/results/nvme0n1/nvme/033.out.bad       2022-06-02 12:03:06.539798439 +0800
    @@ -1,7 +1,10 @@
     Running nvme/033
    -Discovery Log Number of Records 1, Generation counter X
    +Discovery Log Number of Records 2, Generation counter X
     =====Discovery Log Entry 0======
     trtype:  loop
    +subnqn:  nqn.2014-08.org.nvmexpress.discovery
    +=====Discovery Log Entry 1======
    ...
    (Run 'diff -u tests/nvme/033.out /home/lizhijian/blktests/results/nvme0n1/nvme/033.out.bad' to see the entire diff)

```

```
[root@rdma-client blktests]# nvme discover -t loop

Discovery Log Number of Records 2, Generation counter 2
=====Discovery Log Entry 0======
trtype:  loop
adrfam:  ipv4
subtype: unrecognized
treq:    not specified, sq flow control disable supported
portid:  0
trsvcid: 4420
subnqn:  nqn.2014-08.org.nvmexpress.discovery
traddr:  127.0.0.1
=====Discovery Log Entry 1======
trtype:  loop
adrfam:  ipv4
subtype: nvme subsystem
treq:    not specified, sq flow control disable supported
portid:  0
trsvcid: 4420
subnqn:  blktests-subsystem-1
traddr:  127.0.0.1
```

**Comment by yangx-jy** [comment]
I think the following patch can fix the issue:
https://lore.kernel.org/linux-block/20220604160638.1118-1-yangx.jy@fujitsu.com/T/#u

**Comment by zhijianli88** [comment]
> I think the following patch can fix the issue: https://lore.kernel.org/linux-block/20220604160638.1118-1-yangx.jy@fujitsu.com/T/#u

verified, it works.

Fixes by 06619bf630d6839a04c9f0b1792502c90a83853a
